mobile/library/jni/jni_helper.cc

```diff
@@ -10,6 +12,7 @@ constexpr jint JNI_VERSION = JNI_VERSION_1_6;
 constexpr const char* THREAD_NAME = "EnvoyMain";
 std::atomic<JavaVM*> java_vm_cache_;
 thread_local JNIEnv* jni_env_cache_ = nullptr;
+absl::flat_hash_map<absl::string_view, jclass> JCLASS_CACHES;
```

Comment:
how large would this map grow?  I'm concerned about keeping it unbounded in size.

Reply:
We control the cache via `addClassToCache` in the `JNI_OnLoad`. Right now we only cache 2 classes, but in the future it'll be whatever Envoy Mobile needs, but it'll be around 10-20.